src/generate-fallback-map.js

```diff
@@ -0,0 +1,39 @@
+export default function generateFallbackMap(fonts, preloadDepth) {
+  // map of font name to array of possible fallbacks (name and style offsets) (closest match first)
+  const fallbackCandidateLookup = {};
+  // keys are fonts that depth strategy suggests we should preload, values are always true
+  const preloadableFallbacks = {};
+  // keys are font names, values are objects with properties for fallback name and style offsets
+  const fallbackLookup = {};
+
+  // FIRST TRAVERSAL: gather set of possible fallbacks per font and
+  // the list of all fonts our depth strategy suggests we should preload
+  Object.keys(fonts).forEach(fontName => {
```

Comment:
Generally I like this coding style, but I wonder if it would make sense to rewrite this in a more performant way given the fact that this is a performance focused library. For example, only calling `Object.keys` on `fonts` once rather than twice, and using a standard `for` loop rather than `forEach`. Not a requirement, just a thought

Reply:
Will add an issue to re-visit logic with an eye to perf

Reply:
#4